packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -856,6 +902,23 @@ class _RenderSegmentedControl<T extends Object> extends RenderBox
 
   double get totalSeparatorWidth => (_kSeparatorInset.horizontal + _kSeparatorWidth) * (childCount ~/ 2);
 
+  List<double> childWidths = <double>[];
+  int? getSegmentIndex(double dx, TextDirection textDirection) {
+    double subtotalWidth = 0;
+    for (int i = 0; i < childWidths.length; i++) {
+      final double segmentWidth = childWidths[i];
```

Comment:
I think there are separators in between the children?

Reply:
Ah `childWidths` only contains the segment widths. This is a bad name. Renamed it to `segmentWidths`!
https://github.com/flutter/flutter/pull/153125/files#diff-aba8be2635c1c4a4ee9c15b8c0bdb7ba4cb22a4f795d4e98e5cebda3fc10666eR1025